recipes/default.rb

```diff
@@ -221,6 +223,9 @@
   node.override['hopsworks']['reserved_project_names'] = "#{node['hopsworks']['reserved_project_names']},cloud"
 end
 
+# encrypt onlinefs user password
+encrypted_onlinfefs_password = Digest::SHA256.hexdigest node['onlinefs']['hopsworks']['password'] + node['onlinefs']['hopsworks']['salt']
```

Comment:
I assume that the salt needed outside this recipe/cookbook - it's generated externally and stored in a table?
If not, why not just generate a random number?

Reply:
The reason we do this is because we need the clear text password in the onlinefs::default recipe to be able to log in as the user and get an api key for the service. The salt could just be a random string, yes, I just thought it's good to have it defined together with the password in the attributes of onlinefs-chef, and not somewhere in the middle of a recipe.

Reply:
changed it to use random salt